contracts/wrapper/NameWrapper.sol

```diff
@@ -657,6 +644,53 @@ contract NameWrapper is
         emit NameWrapped(node, name, wrappedOwner, fuses);
     }
 
+    function _addLabelAndWrap(
+        bytes32 parentNode,
+        string memory label,
+        address newOwner,
+        uint96 _fuses
+    ) internal {
+        bytes32 labelhash = keccak256(bytes(label));
+        bytes32 node = _makeNode(parentNode, labelhash);
+        bytes memory name = _addLabel(label, names[parentNode]);
+        _wrap(node, name, newOwner, _fuses);
+    }
+
+    function _setSubnodeOwnerAndWrap(
+        bytes32 parentNode,
+        string memory label,
+        address newOwner,
+        uint96 _fuses
+    ) internal {
+        bytes32 labelhash = keccak256(bytes(label));
+        bytes32 node = _makeNode(parentNode, labelhash);
+        bytes memory name = _addLabel(label, names[parentNode]);
+        ens.setSubnodeOwner(parentNode, labelhash, address(this));
+        _wrap(node, name, newOwner, _fuses);
+    }
+
+    function _transferAndBurnFuses(
+        bytes32 parentNode,
+        bytes32 labelhash,
+        address newOwner,
+        uint96 _fuses
+    ) internal {
+        bytes32 subnode = _makeNode(parentNode, labelhash);
+        (, uint96 fuses) = getData(uint256(subnode));
+
+        uint96 newFuses = fuses | _fuses;
+
+        _setData(uint256(subnode), newOwner, newFuses);
```

Comment:
Do you mean burnChildFuses? If so, it means we need to do the modifier checks again, and also it doesn't allow changing the owner, but this would also do the transfer to the newOwner

Reply:
I meant the whole set-and-transfer thing; we definitely have that functionality in other functions. Doing all these operations manually, including emitting events, seems like a code smell.

Reply:
I think this is essentially it.

We have 2 functions that do one of each:
* BurnFuses/burnChildFuses()
* safeTransferFrom()

burnFuses only burns fuses and safeTransfer does the transferring only. So this is a function that does both. I could call burnFuses and safeTransfer instead of creating this function, but it would mean we would be doing the owner checks twice.

We probably can expose this specific function publicly so it can be part of the API.